### ISSUE TITLE
Wally should return error codes when an operation is not successful

### ISSUE THREAD

**Issue by NobleDraconian** [UpliftGames/wally#104]
Currently, as of [v0.3.1](https://github.com/UpliftGames/wally/releases/tag/v0.3.1), wally does not return any error codes when an operation fails. For example, if you run `wally publish`, but there is a version conflict, wally will fail the operation & spit an error message to the console. But it doesn't return an actual error code to the host process!

This makes things like github actions / CI proceed as if the operation were successful, when it wasn't:
![image](https://user-images.githubusercontent.com/26460940/211950506-bdff5419-8ff4-4e6b-87ad-ca1e0add15e9.png)

**Comment by magnalite** [comment]
I agree, adding this should be a quick and easy win for ci/cd workflows
